pkg/cmd/upgrade/upgrade_boot.go

```diff
@@ -521,10 +523,16 @@ func (o *UpgradeBootOptions) deleteLocalBranch(branch string) error {
 
 func (o *UpgradeBootOptions) cloneDevEnv() error {
 	jxClient, devNs, err := o.JXClientAndDevNamespace()
+	if err != nil {
+		return errors.Wrap(err, "failed to create/get jx client and dev namespace")
+	}
 	devEnv, err := kube.GetDevEnvironment(jxClient, devNs)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get dev environment in namespace %s", devNs)
 	}
+	if devEnv == nil {
+		return errors.Wrapf(errors.New("Cannot find environment with dev selector"),"No development environment found for namespace %s", devNs)
```

Comment:
FYI, this is likely to have a lint failure due to the lack of a space after the comma.

Reply:
Fixed.

Reply:
all tests  passed. boot-vault seems to fail in the first attempt, and succeeds in the second, will take a look.